### PR TITLE
feat: fetch zonefile data from bns v2 api

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out Leather build efb409c — [Extension build](https://github.com/leather-io/extension/actions/runs/11855887646), [Test report](https://leather-io.github.io/playwright-reports/feat-bns-v2), [Storybook](https://feat-bns-v2--65982789c7e2278518f189e7.chromatic.com), [Chromatic](https://www.chromatic.com/library?appId=65982789c7e2278518f189e7&branch=feat-bns-v2)

This pr adds fetching zonefile from bns v2

### CODE REVIEW

**Comment by 314159265359879** [comment]
I see 404's (I tested if that was related to a VPN I was using, doesn't look like it)
image

image

**Comment by 314159265359879** [comment]
API says there is no zonefile... for it to get the stacks address mapping it shouldn't need a zonefile right?

https://api.bnsv2.com/resolve-name/werner.btc
`{"error":"No zonefile found"}`

**Review by fbwoolf — APPROVED** [review]

**Review by pete-watters — APPROVED** [review]
Nice work 🎉 

I wonder if we need to always say `BnsV2` if we won't support `v1` anymore? (I think we don't)

So we could just say `bns`

File: src/app/pages/send/send-crypto-asset-form/family/stacks/components/stacks-recipient-field.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { fetchNameOwner } from '@leather.io/query';
+import { fetchStacksNameOwner } from '@leather.io/query';
 
 import { RecipientField } from '../../../components/recipient-fields/recipient-field';
 
 export function StacksRecipientField() {
-  return <RecipientField bnsLookupFn={fetchNameOwner} />;
+  return <RecipientField bnsLookupFn={fetchStacksNameOwner} />;
```
Comment (pete-watters) [inline src/app/pages/send/send-crypto-asset-form/family/stacks/components/stacks-recipient-field.tsx:6]: I took a look into `fetchStacksNameOwner` from curiosity and see that it's almost exactly the same as `fetchBtcNameOwner` but we check for a different field.

Can we:
- make these more like the original `fetchNameOwner` to share code?
- get rid of the old `fetchNameOwner` we don't use now?

Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/family/stacks/components/stacks-recipient-field.tsx:6]: I fixed this, so there are different fetch functions now

**Review by 314159265359879 — APPROVED** [review]
Tested again to working as I expect now. 

Stacks address is based on holder of the BNSv2 name
Bitcoin address is based on what is in the zonefile


Some addresses do not neatly display on one line, should they?
image

**Comment by alter-eggo** [comment]
@314159265359879 don't think this is a big problem, also need to make it way smaller, so btc address and in extension mode it to be also in one line

**Pull request merged**